src/Microsoft.AspNetCore.Razor.Evolution/DefaultRazorRuntimeCSharpLoweringPhase.cs

```diff
@@ -332,6 +332,16 @@ internal override void VisitCreateTagHelper(CreateTagHelperIRNode node)
                     tagHelperVariableName);
             }
 
+            internal override void VisitAddPreallocatedTagHelperHtmlAttribute(AddPreallocatedTagHelperHtmlAttributeIRNode node)
```

Comment:
The intent is that IR nodes have some kind of semantic meaning, not that they stand in for codegen.